### ISSUE TITLE
Messaging Re-design clean up - Remove old messaging flow

### ISSUE THREAD

**Issue by JessicaMarine** [flexion/ef-cms#5511]
As a development team, in order to maintain a clean code base and ensure the changes we made to the messaging flow work, we need to remove all of the old messaging flow items that are not used for the new flow and run E2E tests on the new flow. 

## Pre-Conditions
* #5841/Revamping the stip decision flow

## Acceptance Criteria
* Disable old messaging flow
* Test new messaging flow - E2E
* Remove "Messages" tab from Document QC pages, as well as any other UI remnants from the old flow
* Old document detail page is removed 
* Once we remove "Messages" from the global header, we need to rename "Case Messages" to "Messages" and make sure everything is renamed correctly on the back end. 

## Security Considerations
 - [ ] Does this work make you nervous about privacy or security?
 - [ ] Does this work make major changes to the system?
 - [ ] Does this work implement new authentication or security controls?
 - [ ] Does this work create new methods of authentication, modify existing security controls, or explicitly implement any security or privacy features?

## Tasks
- [x] 1) Remove old Messages link in the page header and replace with Case Message (Mrin & Rae & Rachael #6048)
- [x] 2) Remove old Messages views (Mrin & Rae & Rachael #6048)
  - [x] Remove route
- [x] 3) Remove old Document Detail view (Rachael #6040)
  - [x] Remove route (Rachael #6040)
- [x] 4) Remove old Message entity (Rachael #6042)
- [x] 5) Remove Messages from Work Items (Rachael #6042)
- [x] 6) Remove Messages tab from Document QC pages (Rachael #6043)
- [x] 7) Update all references of `Case Message` to `Message`(Mrinalini) #6060
- [x] 8) Remove WorkItem `isQC` (Rachael #6052)
- [x] 9) Remove `workQueueIsInternal` (Rachael #6050)
- [x] 10) Convert Document.workItems to a single object instead of an array (Rachael #6062)
- [x] 11) Remove createWorkItemInteractor (Rachael #6044)
- [x] 12) Remove SignStipDecision.jsx and route, sequences (Rachael #6045)
- [x] 13) Remove/clean up documentDetailHelper (Rachael #6047)
- [x] 14) Remove/clean up DocumentDetailHeader (Rachael #6047)
- [x] 15) Remove endpoints/persistence methods for old messaging (Rachael #6056)
- [x] 16) Rename CaseMessage* views to Message* (Rachael #6072)
- [x] 17) Rename `/case-message/*` routes to `/message/*` (Rachael #6073)
- [x] 18) Remove code related to old messaging view - CreateMessageModalDialog, ForwardMessage entity (Rachael #6074)
- [x] 19) Update actions, sequences, proxies, interactors, computeds, etc with *CaseMessage* in the name to *Message* (Rachael)

## Definition of Done (Updated 7-2-20)

**Product Owner**
 - [x]  Acceptance criteria have been met
**UX**
 - [x] Business test scenarios to meet all acceptance criteria have been written
 - [x] Usability has been validated
 - [x] Wiki has been updated (if applicable) 
 - [x] Story has been tested on a mobile device (for external users only)

**Engineering**
 - [x] Automated test scripts have been written
 - [x] Field level and page level validation errors (front-end and server-side) integrated and functioning
 - [x] Paired w/Mark on data migration work (if applicable)
 - [x] Verify that language for docket record for internal users and external users is identical
 - [x] New screens have been added to pa11y scripts
 - [x] All new functionality verified to work with keyboard and macOS voiceover https://www.apple.com/voiceover/info/guide/_1124.html 
 - [x] READMEs, other appropriate docs, JSDocs and swagger/APIs fully updated
 - [x] UI should be touch optimized and responsive for external only (functions on supported mobile devices and optimized for screen sizes as required)
 - [x] Module dependencies are up-to-date and are at the latest resolvable version (npm update)
 - [x] Errors in Sonarcloud are fixed https://sonarcloud.io/organizations/flexion-github/projects
 - [x] Lambdas include CloudWatch logging of users, inputs and outputs
 - [x] Code refactored for clarity and to remove any known technical debt
 - [x] Rebuild entity documentation
 - [x] Acceptance criteria for the story has been met
 - [x] Deployed to the dev environment
 - [x] Deployed to the stage environment

**Comment by klohman** [comment]
### Redesign of Correspondence tab
![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/4bced025-ff58-4226-a176-d5f128892afb)

### Redesign of Judges/Chambers Dashboard 

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/f5159765-1163-415a-b2c3-2363a950e5f3)

**Comment by klohman** [comment]
Tests

* Send message without a document attached (https://github.com/flexion/ef-cms/issues/4293)
* Send message with a document attached (https://github.com/flexion/ef-cms/issues/4607)
* Forward message (https://github.com/flexion/ef-cms/issues/4609)
* Reply to message (https://github.com/flexion/ef-cms/issues/4609)
* Complete message (https://github.com/flexion/ef-cms/issues/4609)
* Create Order from message (https://github.com/flexion/ef-cms/issues/4611)
* Apply and Edit Signature to document attached to message (https://github.com/flexion/ef-cms/issues/4612)
* Edit document attached to message (https://github.com/flexion/ef-cms/issues/4612)
* Add Docket Entry for document attached to message (https://github.com/flexion/ef-cms/issues/4612)
* Serve document attached to message (https://app.zenhub.com/workspaces/flexionef-cms-5bbe4bed4b5806bc2bec65d3/issues/flexion/ef-cms/5811)
* View message thread (https://github.com/flexion/ef-cms/issues/4609)
* View completed message thread (https://github.com/flexion/ef-cms/issues/4609)
* Judge's view recent messages on their dashboard (https://app.zenhub.com/workspaces/flexionef-cms-5bbe4bed4b5806bc2bec65d3/issues/flexion/ef-cms/5810)

**Comment by klohman** [comment]
Notes from Jessica (for testing). 

 (1) I don't think the Recent Messages on the Judge's dashboard is working as expected.  I thought it was supposed to show the 5 most recent messages. Check out Judge Buch's -- it's not showing the 5 most recent and they aren't in newest-to-oldest order.  (2) Also, in general, there is a lag in messages being added and removed from the Recent Messages on the Judge's dashboard.  In staging, a DocketClerk1 sent a Proposed Stipulated Decision to jasford.  That message wasn't displaying on the dashboard when I signed in as Judge Ashford.  I had to click around a bit before it showed up.  Then, when Judge Ashford replied to the message, it stayed on the Recent messages until I clicked around a bit again.  (3) Also, when a Docket Clerk completes a message, it's not clearing it from the Inbox until you click around a bit.  All of this messaging behavior has to work as expected.
